### PR TITLE
swap: refactor htlc construction to allow passing of internal keys

### PR DESCRIPTION
**Pull request by bhandras**
This PR is a refactor of how we construct htlcs to make it possible to pass in internal keys for the sender and receiver when creating P2TR htlcs. Furthermore the commit also cleans up constructors to not pass in script versions and output types to make the code more readable.

### CODE REVIEW

File: swap/htlc.go [hunk 1]
```diff
@@ -527,6 +541,31 @@ func newHTLCScriptV3(cltvExpiry int32, senderHtlcKey, receiverHtlcKey [33]byte,
 
 	rootHash := tree.RootNode.TapHash()
 
+	// Parse the pub keys used in the internal aggregate key. The are
```
Comment (GeorgeTsagk) [inline swap/htlc.go:544]: nit: typo "The are" -> "They are"

**Review by GeorgeTsagk — APPROVED** [review]
LGTM 🎉
just 1 small typo

**Review by sputn1ck — APPROVED** [review]
LGTM nice refactor

File: swap/htlc.go (hunk 1 above)
Comment (bhandras) [reply swap/htlc.go]: done

**Pull request merged**